dds/idl/marshal_generator.cpp

```diff
@@ -1968,11 +2063,87 @@ bool marshal_generator::gen_struct(AST_Structure* node,
         cases <<
           "    case " << id << ": {\n"
           "      if (!" << streamCommon(field_name, fields[i]->field_type(),
-            ">> stru", intro, cxx) << ") {\n"
+            ">> stru", intro, cxx) << ") {\n";
```

Comment:
So I was under the impression that `DISCARD` meant the end of deserialization of the entire sample and I was writing a comment about that when it clicked, and I looked at the spec, realized I was mistaken. I think you generally have the right general idea: `DISCARD` will be caught like an exception by `USE_DEFAULT` and `TRIM`, and deserialization can continue. So nothing wrong there.

The only thought I have so far is there should probably be an explicit difference between a discard and a fundamental serialization failure that indicates we can't continue reading the stream. I'm thinking maybe a separate discard status in the `Serializer`? We can discuss it after the stand up if you'd like.